core/domain/exp_domain.py

```diff
@@ -1145,8 +1138,17 @@ def update_interaction_answer_groups(self, answer_groups_list):
                         'Expected rule_inputs to be a dict, received %s'
                         % rule_inputs)
                 for param_name, value in rule_inputs.iteritems():
-                    param_type = rule_domain.get_obj_type_for_param_name(
-                        matched_rule, param_name)
+                    param_list = rule_domain.get_param_list(
+                        matched_rule_description)
+
+                    param_type = None
```

Comment:
Good idea -- done. I added two new functions to extensions/interactions/base.py and removed rule_domain.